### PR TITLE
Dev/arek

### PR DESCRIPTION
**Pull request by arekouzounian**
Just wanted to make a quick PR before I start working on the run command in the CLI so that I don't end up sending one huge PR at the end

### CODE REVIEW

File: cli/runner/cmd/requests.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// Simple struct to hold JSON vals from API call
+type Langs struct {
+	Languages []string `json:"languages"`
+	Error     bool
+}
+
+func getLangListJSON(server string, endpoint string, jsonLangs *Langs) string {
+	resp, err := http.Get(server + endpoint)
+	if err != nil {
+		return "Bad API Call"
```
Comment (camerondurham) [inline cli/runner/cmd/requests.go:21]: Nit: with the comment for line 18, you can return the `error` type for caller to print (like `return nil, err`). It's kind of small but might help to have an error that says a little more specifically what's wrong like this when the server's not running:

```
Get "http://localhost:8080/api/v1/languages": dial tcp 127.0.0.1:8080: connect: connection refused
```

File: cli/runner/cmd/run.go [hunk 2]
```diff
@@ -10,12 +10,21 @@ import (
 	"github.com/spf13/cobra"
 )
 
+/*
+	Example Usage:
+	runner run source.py
+	runner run -l source.py # HAS CLI CHECK FOR LANGUAGE
```
Comment (camerondurham) [inline cli/runner/cmd/run.go:16]: Awesome, thanks for including this! You can actually write this into the `&cobra.Command{}` directly so the user can see this too when they run with `--help` flag.

```go
&cobra.Command{
	Use:   "run",
	Short: "runs the supplied code loaded in from a file.",
    Long: `
Example Usage:
	runner run source.py
	runner run -l source.py # HAS CLI CHECK FOR LANGUAGE
`
    ...
}
```

**Review by camerondurham — COMMENTED** [review]
Looking good! And thanks for sending smaller PR, appreciate it. I left a couple comments on function signature/error handling and a few nit-picks/nice to haves.

**Comment by arekouzounian** [comment]
I think I resolved all the issues you brought up, I just wanted to get some clarification on my most recent commit because I'm wondering whether the error check in requests.go
```
if resp.StatusCode > 299 {
	return nil, fmt.Errorf("Response failed with status code: %d\n", resp.StatusCode)
} else if err != nil {
	return nil, err
}
```
should just be one check. What do you think?

File: cli/runner/cmd/langs.go [hunk 3]
```diff
@@ -10,13 +10,29 @@ import (
...
+)
+
 // langsCmd represents the langs command
 var langsCmd = &cobra.Command{
 	Use:   "langs",
 	Short: "query the server for supported languages",
 	Run: func(cmd *cobra.Command, args []string) {
 		// implement CLI subcommand logic here
-		fmt.Println("langs called")
+		resp, err := getLangListJSON(server, LANG_ENDPOINT)
```
Comment (camerondurham) [inline cli/runner/cmd/langs.go:27]: Nice, thanks for updating this!

File: cli/runner/cmd/requests.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+	body, err := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	//should I just avoid the status code check and just have one single error check here?
+	if resp.StatusCode > 299 {
+		return nil, fmt.Errorf("Response failed with status code: %d\n", resp.StatusCode)
+	} else if err != nil {
+		return nil, err
+	}
```
Comment (camerondurham) [inline cli/runner/cmd/requests.go:33]: I missed commenting this before, sorry. You're right to have both checks here since [http.Get](https://pkg.go.dev/net/http#Client.Get) has like a few failure cases:

1. Error
    1. `err != nil`: there is a `*url.Error`
    2. `err != nil` and `resp.StatusCode` is not 2xx: server returned an error response (don' worry about handling these cases specifically)
        1. 4xx: invalid request
        2. 5xx: server failure, timeout, etc
3. Ok
    1. `err != nil` and `resp.StatusCode` is 2xx (it could be 1xx but idk if anyone uses these codes...)

I just have one final nit pick: could we move the `resp.StatusCode` check before `io.ReadAll`? Since those two checks are handling possible "errors" for the `http.Get` call. It's not wrong to have it where it is but I personally think it makes more sense right after `http.Get`.

Thanks!

**Review by camerondurham — COMMENTED** [review]
Great! Thanks for the changes! I'm sorry I have one other minor comment about moving error checking slightly.

TLDR: you're right to have both error checks because of how [http.Get](https://pkg.go.dev/net/http#Client.Get) can fail. I just had a minor comment on the sequence of error checks.

**Review by camerondurham — APPROVED** [review]
LGTM!
